bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -720,8 +720,7 @@ public void asyncAddEntry(final byte[] data, final AddCallback cb,
      * @param ctx
      *            some control object
      */
-    public void asyncAddEntry(final long entryId, final byte[] data, final AddCallback cb, final Object ctx)
-            throws BKException {
+    public void asyncAddEntry(final long entryId, final byte[] data, final AddCallback cb, final Object ctx) {
```

Comment:
@sijie as LedgerHandleAdv is a subclass I thought it was automatic that all the simple API should work as in the normal LedgerHandle. If this is this the case then I am ok with making all functions without explicit entryid fail. I have not tested with 4.4.
IMHO if possible it would be better to support the full API, I think that as last option we will need an extra boolean field in PendingAddOp